components/job-orchestration/job_orchestration/garbage_collector/garbage_collector.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python3
+
+import argparse
+import asyncio
+import os
+import sys
+from pathlib import Path
+from typing import Callable, Dict, List, Optional, Tuple
+
+from clp_py_utils.clp_config import (
+    CLPConfig,
+    GARBAGE_COLLECTOR_NAME,
+)
+from clp_py_utils.clp_logging import get_logger
+from clp_py_utils.core import read_yaml_config_file
+from job_orchestration.garbage_collector.archive_garbage_collector import archive_garbage_collector
+from job_orchestration.garbage_collector.constants import (
+    ARCHIVE_GARBAGE_COLLECTOR_NAME,
+    SEARCH_RESULT_GARBAGE_COLLECTOR_NAME,
+)
+from job_orchestration.garbage_collector.search_result_garbage_collector import (
+    search_result_garbage_collector,
+)
+from job_orchestration.garbage_collector.utils import configure_logger
+from pydantic import ValidationError
+
+logger = get_logger(GARBAGE_COLLECTOR_NAME)
+
+
+async def main(argv: List[str]) -> int:
+    args_parser = argparse.ArgumentParser(description=f"Spin up the {GARBAGE_COLLECTOR_NAME}.")
+    args_parser.add_argument("--config", "-c", required=True, help="CLP configuration file.")
+    parsed_args = args_parser.parse_args(argv[1:])
+
+    # Setup logging to file
+    logs_directory = Path(os.getenv("CLP_LOGS_DIR"))
+    logging_level = os.getenv("CLP_LOGGING_LEVEL")
+    configure_logger(logger, logging_level, logs_directory, GARBAGE_COLLECTOR_NAME)
+
+    # Load configuration
+    config_path = Path(parsed_args.config)
+    try:
+        clp_config = CLPConfig.parse_obj(read_yaml_config_file(config_path))
+    except ValidationError as err:
+        logger.error(err)
+        return 1
+    except Exception as ex:
+        logger.error(ex)
```

Comment:
```suggestion
    except Exception:
        logger.exception("Failed to parse CLP configuration file.")
```
We can log like this for exceptions, right?